src/lr3_comp.erl

```diff
@@ -52,26 +47,36 @@ compile_normal_apps(State) ->
 
 compile_normal_app(AppInfo) ->
     lr3_comp_util:copy_app_src(AppInfo),
-    Opts = rebar_app_info:opts(AppInfo),
-    AppDir = rebar_app_info:dir(AppInfo),
+    Opts         = rebar_app_info:opts(AppInfo),
+    AppDir       = rebar_app_info:dir(AppInfo),
     OtherSrcDirs = rebar_dir:src_dirs(Opts),
+    SourceDirs   = lr3_comp_util:get_src_dirs(AppDir, ["src"] ++ OtherSrcDirs),
+    OutDir       = lr3_comp_util:relative_out_dir(AppInfo),
+    FirstFiles   = lr3_comp_util:get_first_files(Opts, AppDir),
+    ErlOpts      = rebar_opts:erl_opts(Opts),
+    Config       = config(OutDir, ErlOpts),
     rebar_api:debug("\tOtherSrcDirs: ~p", [OtherSrcDirs]),
-    SourceDirs = lr3_comp_util:get_src_dirs(AppDir, ["src"] ++ OtherSrcDirs),
-    %%OutDir = 'lfe-compiler-util':out_dir(AppDir),
-    OutDir = lr3_comp_util:relative_out_dir(AppInfo),
-    FirstFiles = lr3_comp_util:get_first_files(Opts, AppDir),
-    Files = lr3_comp_util:get_files(FirstFiles, SourceDirs),
     rebar_api:debug("\tAppInfoDir: ~p", [AppDir]),
     rebar_api:debug("\tSourceDirs: ~p", [SourceDirs]),
     rebar_api:debug("\tOutDir: ~p", [OutDir]),
     rebar_api:debug("\tFirstFiles: ~p", [FirstFiles]),
-    rebar_api:debug("\tFiles: ~p", [Files]),
-    DoCompile = fun(Source, Opts1) ->
-                  rebar_api:console(" ~~~~> \tCompiling ~s ...",
-                                    [lr3_comp_util:relative(Source)]),
-                  compile(Opts1, Source, AppDir, OutDir)
-                end,
-    rebar_base_compiler:run(Opts, [], Files, DoCompile),
+    rebar_api:debug("\tErlOpts: ~p", [ErlOpts]),
+    CompileDir = fun(Dir) -> compile_dir(Config, FirstFiles, Dir, OutDir) end,
+    lists:foreach(CompileDir, SourceDirs),
     rebar_api:debug("\tFinished compile.", []),
     lr3_comp_util:copy_beam_files(AppInfo, OutDir),
     code:add_patha(lr3_comp_util:out_dir(rebar_app_info:dir(AppInfo))).
+
+-spec config(file:dirname(), list()) -> list().
+config(OutDir, ErlOpts) ->
+    [{outdir, OutDir}] ++ ErlOpts ++
+        [{i, lr3_comp_util:include_dir()}, return, verbose].
+
+-spec ensure_dir(file:dirname()) -> ok.
+ensure_dir(OutDir) ->
+    %% Make sure that ebin/ exists and is on the path
+    ok = filelib:ensure_dir(filename:join(OutDir, "dummy.beam")),
+    AbsOutDir = filename:absname(OutDir),
+    rebar_api:debug("\t\tAdding ~p to path ...", [AbsOutDir]),
+    true = code:add_patha(AbsOutDir),
+    ok.
```

Comment:
Yeah, go ahead and put those in ``lr3_comp_util.erl``, and I'll merge your PR :-)

Thanks!